scripts/stubsabot.py

```diff
@@ -470,11 +499,91 @@ async def analyze_diff(
     return DiffAnalysis(py_files=py_files, py_files_stubbed_in_typeshed=py_files_stubbed_in_typeshed)
 
 
-async def determine_action(distribution: str, session: aiohttp.ClientSession) -> Update | NoUpdate | Obsolete:
+def _add_months(date: datetime.date, months: int) -> datetime.date:
+    month = date.month - 1 + months
+    year = date.year + month // 12
+    month = month % 12 + 1
+    day = min(date.day, calendar.monthrange(year, month)[1])
+    return datetime.date(year, month, day)
+
+
+def obsolete_more_than_6_months(distribution: str) -> bool:
+    try:
+        with metadata_path(distribution).open("rb") as file:
+            data = tomlkit.load(file)
+    except FileNotFoundError:
+        raise NoSuchStubError(f"Typeshed has no stubs for {distribution!r}!") from None
+
+    obsolete_since = data["obsolete_since"]
+    if not obsolete_since:
+        return False
+
+    assert type(obsolete_since) is String
+    comment: str | None = obsolete_since.trivia.comment
+    if not comment:
+        return False
+
+    release_date_string = comment.removeprefix("# Released on ")
```

Comment:
Note: This is probably also something that we should put into metadata (possibly as a structured `obsolete` field with `since_version` and `since_date` fields), but again in a later PR.